docs/feature_midi.md

```diff
@@ -0,0 +1,254 @@
+# MIDI
+
+## Usage
+
+First, enable MIDI by adding the following to your `rules.mk`:
+
+```makefile
+MIDI_ENABLE = yes
+```
+
+There are two MIDI systems in QMK: basic and advanced. With basic MIDI you will only be able to send Note On and Note Off messages using the note keycodes, meaning that keycodes like `MI_OCTU` and `MI_OCTD` will not work. Advanced MIDI allows you to do things like octave shifts, channel changes, velocity changes, modulation, and more.
+
+### Basic MIDI
+
+To enable basic MIDI, add the following to your `config.h`:
+
+```c
+#define MIDI_BASIC
+```
+
+### Advanced MIDI
+
+To enable advanced MIDI, add the following to your `config.h`:
+
+```c
+#define MIDI_ADVANCED
+```
+
+#### Sending MIDI Control Codes (MIDI CC)
+
+If you're aiming to emulate the features of something like a Launchpad or other MIDI controller you'll need to access the internal MIDI device directly.
+
+When sending MIDI CC you don't get the advantages of a preimplemented keycode and you will need to implement custom keycodes if you want to use them in your keymap directly using the function `process_record_user`. It is the same process as implementing custom keycodes for macros.
+
+For an overview of that process look at: [Macros](feature_macros.md)
+
+1. First enable MIDI_ADVANCED
+2. use an `extern MidiDevice midi_device;` statement to bring the MIDI device into scope.
+3. Send a control code with the `midi_send_cc(*midi_device, channel, number, value)` function
```

Comment:
```suggestion
```
This is more or less achieved by having this section under the MIDI Advanced section, and the other two points should be self explanatory from the example code below.